examples/prom/maxwell_local_rom_greedy.cpp

```diff
@@ -0,0 +1,725 @@
+        //    an error indicator at a new point.)
+        if (build_database)
+        {
+            double local_rom_freq = 0.0;
+            double curr_freq = 0.0;
+            struct CAROM::GreedyErrorIndicatorPoint pointRequiringRelativeError =
+                greedy_sampler->getNextPointRequiringRelativeError();
+            CAROM::Vector* relativeErrorPointData = pointRequiringRelativeError.point.get();
+            struct CAROM::GreedyErrorIndicatorPoint pointRequiringErrorIndicator =
+                greedy_sampler->getNextPointRequiringErrorIndicator();
+            CAROM::Vector* errorIndicatorPointData =
+                pointRequiringErrorIndicator.point.get();
+
+            if (relativeErrorPointData != NULL)
+            {
+                if (myid == 0) cout << "Calculating the relative error of "
+                                        << "the last sampled point." << endl;
+
+                local_rom_freq = pointRequiringRelativeError.localROM.get()->item(0);
+                curr_freq = pointRequiringRelativeError.point.get()->item(0);
+                if (myid == 0) cout << "Using the basis obtained at the frequency: " <<
+                                        local_rom_freq << endl;
+                online = true;
+                calc_rel_error = true;
+            }
+            else if (errorIndicatorPointData != NULL)
+            {
+                if (myid == 0) cout << "Calculating an error indicator at "
+                                        << "a new point." << endl;
+
+                local_rom_freq = pointRequiringErrorIndicator.localROM.get()->item(0);
+                curr_freq = pointRequiringErrorIndicator.point.get()->item(0);
+                if (myid == 0) cout << "Using the basis obtained at the frequency: "
+                                        << local_rom_freq << endl;
+                online = true;
+                calc_err_indicator = true;
+            }
+            else
+            {
+                std::shared_ptr<CAROM::Vector> nextSampleParameterPoint =
+                    greedy_sampler->getNextParameterPoint();
+                CAROM::Vector* samplePointData = nextSampleParameterPoint.get();
+                if (samplePointData != NULL)
+                {
+                    if (myid == 0) cout << "Sampling a new point." << endl;
+                    local_rom_freq = samplePointData->item(0);
+                    curr_freq = samplePointData->item(0);
+                    offline = true;
+                }
+                else
+                {
+                    if (myid == 0) cout << "The greedy algorithm has finished." << endl;
+                    greedy_sampler->save("maxwell_local_rom_greedy_algorithm_data");
+                    build_database = false;
+                    continue;
+                }
+            }
+            // 4b. Set the correct frequency as commanded by the greedy algorithm.
```

Comment:
What does "correct" mean here?

Reply:
This is a good point. I think the comment is more clear by just deleting the word "correct". The greedy algorithm is the only thing determining new frequency values.